### PR TITLE
RavenDB-17994 before doing validation of private key of the cert to avoid issues with bouncy castle not supporting encryption algorithm used we should export it

### PR DESCRIPTION
**Pull request by ppekrol**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-17994

### Type of change

- Bug fix

### How risky is the change?

- High

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- It has been verified by manual testing

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Server/ServerWide/SecretProtection.cs [hunk 1]
```diff
@@ -669,6 +677,9 @@ public RavenServer.CertificateHolder LoadCertificateFromPath(string path, string
                 // we need to load it as exportable because we might need to send it over the cluster
                 var loadedCertificate = new X509Certificate2(rawData, password, X509KeyStorageFlags.Exportable | X509KeyStorageFlags.MachineKeySet);
 
+                // make sure that we are using supported encryption format by bouncy castle
+                rawData = loadedCertificate.Export(X509ContentType.Pfx, password);
```
Comment (ayende) [inline src/Raven.Server/ServerWide/SecretProtection.cs]: Shouldn't this be:

```suggestion
                rawData = loadedCertificate.Export(X509ContentType.Pfx);
```

Comment (ppekrol) [reply src/Raven.Server/ServerWide/SecretProtection.cs]: What's the difference?

Comment (ayende) [reply src/Raven.Server/ServerWide/SecretProtection.cs]: You are exporting encrypted with password or not. In memory, we just want to use the unencrypted one. Simplify things

**Comment by arekpalinski** [comment]
We haven't seen any issue like that for years. Closing the PR.

**Pull request closed without merging**